iceoryx_posh/source/roudi/memory/mempool_collection_memory_block.cpp

```diff
@@ -36,15 +36,17 @@ MemPoolCollectionMemoryBlock::~MemPoolCollectionMemoryBlock() noexcept
 
 uint64_t MemPoolCollectionMemoryBlock::size() const noexcept
 {
-    return cxx::align(sizeof(mepoo::MemoryManager), mepoo::MemPool::MEMORY_ALIGNMENT)
+    return cxx::align(sizeof(mepoo::MemoryManager),
+        static_cast<unsigned int>(mepoo::MemPool::MEMORY_ALIGNMENT))
            + mepoo::MemoryManager::requiredFullMemorySize(m_memPoolConfig);
 }
 
 uint64_t MemPoolCollectionMemoryBlock::alignment() const noexcept
 {
     // algorithm::align doesn't like constexpr values
     auto memPoolAlignment = mepoo::MemPool::MEMORY_ALIGNMENT;
-    return algorithm::max(alignof(mepoo::MemoryManager), memPoolAlignment);
+    return algorithm::max(alignof(mepoo::MemoryManager),
+        static_cast<unsigned int>(memPoolAlignment));
```

Comment:
Is this tested on mac os and linux arm 32-bit? Does it compile and work there? If not we should definitely do this before merging this!